packages/go_router/lib/src/match.dart

```diff
@@ -61,6 +61,29 @@ class RouteMatch {
     throw MatcherError('Unexpected route type: $route', restLoc);
   }
 
+  /// Completes the promise returned by [GoRouter.push], allowing comunication
+  /// between pages.
+  ///
+  /// If the promise has already been completed, this method does nothing.
```

Comment:
promise is not a keyword in dart. I think you meant `future`. Eitherway, this is implementation detail, we should focus on when/why this method is called.